### PR TITLE
feat: add native logging with headers and body to CloudEvent

### PR DESCRIPTION
**Pull request by grant**
Add better native logging to the `CloudEvent` class.

## Proposed Changes

Logging a `CloudEvent` should call the `toString` of the `CloudEvent`.

Fixes https://github.com/cloudevents/sdk-javascript/issues/432

### CODE REVIEW

File: src/event/cloudevent.ts [hunk 1]
```diff
@@ -195,4 +195,12 @@ See: https://github.com/cloudevents/spec/blob/v1.0/spec.md#type-system`);
   ): CloudEvent {
     return new CloudEvent(Object.assign({}, this.toJSON(), options) as CloudEvent, strict);
   }
+
+  /**
+   * The native `console.log` value of the CloudEvent.
+   * @return The string representation of the CloudEvent.
+   */
+  [Symbol.for("nodejs.util.inspect.custom")]() {
```
Comment (lance) [inline src/event/cloudevent.ts:203]: This is a pretty clever approach. :)

File: test/integration/cloud_event_test.ts [hunk 2]
```diff
@@ -55,6 +55,11 @@ describe("A CloudEvent", () => {
     expect(new CloudEvent(JSON.parse(JSON.stringify(ce)))).to.deep.equal(ce);
   });
 
+  it("serializes as JSON with raw log", () => {
+    const ce = new CloudEvent({ ...fixture, data: { lunch: "tacos" } });
+    expect(ce.toString()).to.deep.equal(JSON.stringify(ce));
```
Comment (lance) [inline test/integration/cloud_event_test.ts:60]: I don't think this is actually testing the change you made. This test should pass, even without the change. E.g. this is from my local repo at `main`.

```
> ce = new CloudEvent({ source: 'test', type: 'test', data: { lunch: 'tacos' } });
CloudEvent {
  id: '7ba9e813-9bd3-4bb1-8323-4e93d8bbe4c4',
  time: '2021-09-28T21:04:04.422Z',
  type: 'test',
  source: 'test',
  specversion: '1.0',
  datacontenttype: undefined,
  subject: undefined,
  datacontentencoding: undefined,
  dataschema: undefined,
  data_base64: undefined,
  schemaurl: undefined
}
> ce.toString()
'{"id":"7ba9e813-9bd3-4bb1-8323-4e93d8bbe4c4","time":"2021-09-28T21:04:04.422Z","type":"test","source":"test","specversion":"1.0","data":{"lunch":"tacos"}}'
> JSON.stringify(ce)
'{"id":"7ba9e813-9bd3-4bb1-8323-4e93d8bbe4c4","time":"2021-09-28T21:04:04.422Z","type":"test","source":"test","specversion":"1.0","data":{"lunch":"tacos"}}'
```

I think a better test might be something like this.
```suggestion
    expect(ce.toString()).to.deep.equal(ce[Symbol.for("nodejs.util.inspect.custom")]());
```

**Review by lance — COMMENTED** [review]
Thanks for making this change, @grant - just a small request re: the test

File: src/event/cloudevent.ts (hunk 1 above)
Comment (grant) [reply src/event/cloudevent.ts:203]: Thanks! Came from: https://nodejs.org/api/util.html#util_util_inspect_custom

**Comment by grant** [comment]
I'm getting errors with the index type of the symbol. Want to go with the existing test and perhaps you can add changes if appropriate after?

**Comment by lance** [comment]
> I'm getting errors with the index type of the symbol. Want to go with the existing test and perhaps you can add changes if appropriate after?

I was able to get the test to pass like this:

```js
  it("serializes as JSON with raw log", () => {
    const ce = new CloudEvent({ ...fixture, data: { lunch: "tacos" } });
    const sym = (Symbol.for("nodejs.util.inspect.custom") as unknown) as string;
    const f = (ce[sym] as CallableFunction).bind(ce);
    expect(ce.toString()).to.deep.equal(f());
  });
```
Also, to eliminate the warnings that are being generated, the CE function should look like this:

```js
  /**
   * The native `console.log` value of the CloudEvent.
   * @return {string} The string representation of the CloudEvent.
   */
  [Symbol.for("nodejs.util.inspect.custom")](): string {
    return this.toString();
  }
```

**Comment by grant** [comment]
Great, thanks @lance. PTAL.

**Review by lance — APPROVED** [review]
LGTM

**Review by lholmquist — APPROVED** [review]

**Pull request merged**